### PR TITLE
feat: sasl auth impl

### PR DESCRIPTION
**Pull request by toondaey**
Closes #

Describe your proposed changes here.

- [x] I've read the contributing section of the project [CONTRIBUTING.md](https://github.com/influxdata/rskafka/blob/main/CONTRIBUTING.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

**Comment by crepererum** [comment]
I see the commit msg checker fails here. You can probably squash everything into a single commit and mention the other two authors in the new "mega commit. See https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors on how to do this.

File: README.md [hunk 1]
```diff
@@ -131,7 +131,7 @@ $ docker-compose -f docker-compose-kafka.yml up
 in one session, and then run:
 
 ```console
-$ TEST_INTEGRATION=1 TEST_BROKER_IMPL=kafka KAFKA_CONNECT=localhost:9011 cargo test
+$ TEST_INTEGRATION=1 TEST_BROKER_IMPL=kafka TEST_DELETE_RECORDS=1 KAFKA_CONNECT=localhost:9011 KAFKA_SASL_CONNECT=localhost:9097 cargo test
```
Comment (crepererum) [inline README.md:134]:
```suggestion
$ TEST_INTEGRATION=1 TEST_BROKER_IMPL=kafka KAFKA_CONNECT=localhost:9011 KAFKA_SASL_CONNECT=localhost:9097 cargo test
```

This is a leftover an old test framework. The features to test are now auto-detected based on `TEST_BROKER_IMPL`.

File: src/connection/transport/sasl.rs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#[derive(Debug, Clone)]
+pub enum SaslConfig {
+    Plain { username: String, password: String },
+}
+
+impl SaslConfig {
+    pub(crate) fn auth_bytes(&self) -> Vec<u8> {
+        match self {
+            Self::Plain { username, password } => {
+                let mut auth: Vec<u8> = vec![0];
```
[the remark this answers is not in the document]
Comment (toondaey) [reply src/connection/transport/sasl.rs:14]: This makes so much sense.

**Comment by toondaey** [comment]
All updated now.

**Comment by toondaey** [comment]
> thank you 🙂

You're welcome! Not to be selfish in anyway, but I wanted to ask if there's a release cycle for this package?

**Comment by crepererum** [comment]
> You're welcome! Not to be selfish in anyway, but I wanted to ask if there's a release cycle for this package?

On demand, I try to get a release out in the next couple of days.

**Comment by toondaey** [comment]
> > You're welcome! Not to be selfish in anyway, but I wanted to ask if there's a release cycle for this package?
> 
> On demand, I try to get a release out in the next couple of days.

Ok. Thanks! Looking forward to the next release.😀

**Comment by crepererum** [comment]
@toondaey 0.5.0 is released ✅

**Comment by Neustradamus** [comment]
Good job!

Linked to:
- https://github.com/scram-sasl/info/issues/1
- @ramrengaswamy: https://github.com/influxdata/rskafka/pull/129
- @seijik42: https://github.com/influxdata/rskafka/pull/198
- @toondaey: https://github.com/influxdata/rskafka/pull/216

**Pull request merged**